src/WireMock.Net/Settings/WireMockServerSettingsParser.cs

```diff
@@ -86,53 +108,63 @@ public static bool TryParseArguments(string[] args, [NotNullWhen(true)] out Wire
                 {
                     settings.Logger = logger;
                 }
-                break;
-        }
 
-        if (parser.Contains(nameof(WireMockServerSettings.Port)))
-        {
-            settings.Port = parser.GetIntValue(nameof(WireMockServerSettings.Port));
-        }
-        else if (settings.HostingScheme is null)
-        {
-            settings.Urls = parser.GetValues("Urls", new[] { "http://*:9091/" });
+                break;
         }
+    }
 
+    private static void TryParseProxyAndRecordSettings(WireMockServerSettings settings, SimpleCommandLineParser parser)
+    {
         var proxyUrl = parser.GetStringValue("ProxyURL") ?? parser.GetStringValue("ProxyUrl");
         if (!string.IsNullOrEmpty(proxyUrl))
         {
             settings.ProxyAndRecordSettings = new ProxyAndRecordSettings
             {
                 AllowAutoRedirect = parser.GetBoolValue("AllowAutoRedirect"),
-                ClientX509Certificate2ThumbprintOrSubjectName = parser.GetStringValue("ClientX509Certificate2ThumbprintOrSubjectName"),
+                ClientX509Certificate2ThumbprintOrSubjectName =
+                    parser.GetStringValue("ClientX509Certificate2ThumbprintOrSubjectName"),
                 ExcludedCookies = parser.GetValues("ExcludedCookies"),
                 ExcludedHeaders = parser.GetValues("ExcludedHeaders"),
                 // PreferProxyMapping = parser.GetBoolValue(nameof(ProxyAndRecordSettings.PreferProxyMapping)),
                 SaveMapping = parser.GetBoolValue("SaveMapping"),
                 SaveMappingForStatusCodePattern = parser.GetStringValue("SaveMappingForStatusCodePattern", "*"),
                 SaveMappingToFile = parser.GetBoolValue("SaveMappingToFile"),
-                UseDefinedRequestMatchers = parser.GetBoolValue(nameof(ProxyAndRecordSettings.UseDefinedRequestMatchers)),
-                AppendGuidToSavedMappingFile = parser.GetBoolValue(nameof(ProxyAndRecordSettings.AppendGuidToSavedMappingFile)),
+                UseDefinedRequestMatchers =
+                    parser.GetBoolValue(nameof(ProxyAndRecordSettings.UseDefinedRequestMatchers)),
+                AppendGuidToSavedMappingFile =
+                    parser.GetBoolValue(nameof(ProxyAndRecordSettings.AppendGuidToSavedMappingFile)),
                 Url = proxyUrl!,
                 SaveMappingSettings = new ProxySaveMappingSettings
                 {
                     StatusCodePattern = parser.GetStringValue("SaveMappingForStatusCodePattern", "*"),
                     // HttpMethods = new ProxySaveMappingSetting<string[]>(parser.GetValues("DoNotSaveMappingForHttpMethods", new string[0]), MatchBehaviour.RejectOnMatch)
+                },
+                ReplaceSettings = new ProxyUrlReplaceSettings

```

Comment:
this code block is not needed anymore, you added the method TryParseProxyUrlReplaceSettings